### ISSUE TITLE
Function return type "None" is incompatible with type "_T@submit" in v2024.4.1

### ISSUE THREAD

**Issue by jackjyq** [microsoft/pyright#7606]
## Environment data

-   Language Server version: v2024.4.1
-   OS and version: Windows 11 22631.3296
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.10.11
-   VSCode settings

```json
{
  "python.analysis.typeCheckingMode": "basic",
  "python.languageServer": "Pylance"
}
```

## Code Snippet

```python
from concurrent.futures import ThreadPoolExecutor


def worker() -> None:
    return

class ModThreadPoolExecutor(ThreadPoolExecutor):
    def submit(self, fn, /, *args, **kwargs):
        return super().submit(fn, *args, **kwargs)


with ModThreadPoolExecutor() as executor:
    executor.submit(worker)
```

## Repro Steps

1. Open VSCode and paste the Code Snippet

## Expected behavior

no type error

## Actual behavior

Argument of type "() -> None" cannot be assigned to parameter "fn" of type "(**_P@submit) -> _T@submit" in function "submit"
  Type "() -> None" cannot be assigned to type "(**_P@submit) -> _T@submit"
    Function return type "None" is incompatible with type "_T@submit"
      Type "None" cannot be assigned to type "_T@submit"Pylance[reportArgumentType](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#reportArgumentType)

## Logs

```
XXX
```

## Others

- Pylance v2024.3.2 does not report this issue 
- [mypy](https://mypy-play.net/?mypy=latest&python=3.12) does not report this issue

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.357, which I just published.
